cmd/kubeadm/app/checks/checks.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checks
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+)
+
+// PreFlightCheck validates the state of the system to ensure kubeadm will be
+// successful as often as possilble.
+type PreFlightCheck interface {
+	Check() (warnings []string, errors []string)
+}
+
+// ServiceCheck verifies that the given service is enabled and active. If we do not
+// detect a supported init system however, all checks are skipped and a warning is
+// returned.
+type ServiceCheck struct {
+	service string
+}
+
+func (sc ServiceCheck) Check() (warnings []string, errors []string) {
+
+	initSystem := getInitSystem()
+	if initSystem == nil {
+		return []string{"no supported init system detected, skipping service checks"}, nil
+	}
+
+	warnings = []string{}
+
+	if !initSystem.ServiceExists(sc.service) {
```

Comment:
My thinking here was to not block someone using systemd, but not for kubelet. So if we see a kubelet service and it's not active, that's an error, but if we see systemd and no kubelet service, it's not obviously not active, but it should be an error instead of a warning as this is very close to the non-systemd use case. Does that help clarify or should we just axe it and assume systemd if present, is managing kubelet, and if it's not active that's an error.